KSPCommunityFixes/Performance/OptimisedVectorLines.cs

```diff
@@ -0,0 +1,261 @@
+            // https://docs.unity3d.com/ScriptReference/Camera.ScreenToWorldPoint.html
+
+            // Normalized device coordinates.
+            Vector4 clipSpace = new Vector4()
+            {
+                x = (screenPosition.x - m_Viewport.x) / m_Viewport.width * 2f - 1f,
+                y = (screenPosition.y - m_Viewport.y) / m_Viewport.height * 2f - 1f,
+                z = 1f,
+                w = screenPosition.z,
+            };
+
+            // Perspective multiplication moves it to clip space.
+            clipSpace.x *= clipSpace.w;
+            clipSpace.y *= clipSpace.w;
+            clipSpace.z *= clipSpace.w;
+
+            Vector4 viewSpace = m_projectionInv * clipSpace;
+            viewSpace.w = 1f;
+            Vector4 worldSpace = m_worldToCameraInv * viewSpace;
+
+            return worldSpace;
+        }
+
+        public static Vector3 ScreenToViewportPoint(Camera camera, Vector3 position)
+        {
+            //if (!patchEnabled)
+            //    return camera.ScreenToViewportPoint(position);
+
+            if (cacheDirty)
+                UpdateCache();
+
+            // Not used by VectorLine.
+
+            return Vector3.zero;
+        }
+
+        private static void UpdateCache()
+        {
+            Camera camera = VectorLine.cam3D;
+
+            m_WorldToClip = camera.projectionMatrix * camera.worldToCameraMatrix;
+            //m_WorldToClipInverse = m_WorldToClip.inverse; // We would use this if I knew how to do ScreenToWorldPoint properly.
+            m_Viewport = camera.pixelRect;
+
+            // Instead we use these.
+            m_worldToCameraInv = camera.worldToCameraMatrix.inverse;
+            m_projectionInv = camera.projectionMatrix.inverse;
+
+            cacheDirty = false;
+        }
+
+        private static Vector3 WorldToClip(Matrix4x4 m, Vector3 point)
+        {
+            // Skip z and use result.z as w.
+
+            Vector3 result = default;
+            result.x = m.m00 * point.x + m.m01 * point.y + m.m02 * point.z + m.m03;
+            result.y = m.m10 * point.x + m.m11 * point.y + m.m12 * point.z + m.m13;
+            result.z = m.m30 * point.x + m.m31 * point.y + m.m32 * point.z + m.m33;
+            float num = 1f / result.z;
+            result.x *= num;
+            result.y *= num;
+            return result;
+        }
+    }
+
+    // todo: How to make this obey Settings.cfg?
+
+    [KSPAddon(KSPAddon.Startup.MainMenu, true)]
+    public class VectorLineFrameCounter : MonoBehaviour
+    {
+        protected void Awake()
+        {
+            DontDestroyOnLoad(this);
+        }
+
+        protected void Update()
+        {
+            // I think this is faster than it would be to compare Time.frameCount to a cached frame count in every call to VectorLineCameraProjection.
+
+            VectorLineCameraProjection.cacheDirty = true;
+        }
+    }
```

Comment:
Better try to avoid using specialized KSPAddons whenever possible.
The main KSPCommunityFixes `KSPAddon` already has a static fixed frame counter `long` incremented from its `FixedUpdate()`, you can add an `Update()` there incrementing a `long` frame counter and use that to compare to the last cached frame. The performance difference between comparing a long and a bool is really insignificant here.